File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        /**
        HashMap solution
        */

        /**
        HashMap<Integer, Integer> hm = new HashMap<>();

        for(int i=0;i<numbers.length;i++) {
            int diff = target - numbers[i];
            if(hm.containsKey(diff)) {
                return new int[]{Math.min(i, hm.get(diff)) + 1, Math.max(i, hm.get(diff)) + 1};
            } else {
                hm.put(numbers[i], i);
            }
        }
        */

        /**
        Two ptr solution
        */
        int left = 0;
        int right = numbers.length - 1;

        while(left < right) {
            int intSum = numbers[left] + numbers[right];
            if(intSum == target) {
                return new int[]{left + 1, right + 1};
            } else if(intSum > target) {
                right--;
            } else {
                left++;
            }
        }

        return new int[]{-1, -1};
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-2.cpp
#include <cctype>

class Solution {
public:
    bool isPalindrome(string s) {
        int n = s.size();
        int left = 0;
        int right = n - 1;
        while(left < right) {
            while (!isalnum(s[left]) && left < right) {
                left += 1;
            }

            while(!isalnum(s[right]) && left < right) {
                right -= 1;
            }

            // if (left >= right) {
            //     break;
            // }

            if (tolower(s[left]) != tolower(s[right])){
                cout << s[left] << " " << s[right] << endl;
                return false;
            }
            left += 1;
            right -= 1;
        }

        return true;

    }
};
